### PR TITLE
Add frontend checks and guide

### PR DESCRIPTION
**Pull request by GabriFila**
# Description

This PR includes the necessary checks and automation to ensure proper lifting and formatting of the code:
- prettier
- ESLint
- husky to run checks before local commit
- github action to perform check

And adds general guidelines for newcomers

### CODE REVIEW

**Comment by GabriFila** [comment]
deploy-staging

File: frontend/.husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+ # shellcheck disable=SC1090
```
Comment (giorio94) [inline frontend/.husky/pre-commit:2]:
```suggestion
# shellcheck disable=SC1090
```

Nit

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -81,3 +81,29 @@ jobs:
...
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          persist-credentials: false
+
+      - name: Setup nodeJS
+        uses: actions/setup-node@v1
+        with:
+          node-version: 12
```
Comment (giorio94) [inline .github/workflows/lint.yml:101]: The dockerfile is using node:14, should this also be updated?

File: frontend/src/includeTailwind.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/* This file serve the only purpose to not include the tailwind import in the less file, to allow its use in storybook */
```
Comment (giorio94) [inline frontend/src/includeTailwind.css:1]:
```suggestion
/* This file serves the only purpose to not include the tailwind import in the less file, to allow its use in storybook */
```

File: frontend/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+## Structure
+
+Our frontend is a [React](https://it.reactjs.org/) application. We use [antd](https://ant.design/) as the main component library and [Tailwind](https://tailwindcss.com/) utilities to handle specific css scenarios (padding and margin). We also use [Storybook](https://storybook.js.org/) to ease teamwork.
+
+The application is made to be deployed using docker and can be hosted on a custom subroute. The application takes some environment variables, each needs to have the `REACT_APP_CROWNLABS` prefix. During local development env variables can be set on the local machine or using a `.env` file. In order to define environment variables at container run-time they need to be defined on the `window` object in the `public/config.js` file.
+
+## CI checks
+
+We use ESLint to enforce linting and code quality. We use Prettier to have a uniform code style. These checks are enforced by a GitHub Action run on PRs. Also locally a pre-commit hook written with [husky](https://typicode.github.io/husky/#/) users [lint-staged](https://github.com/okonet/lint-staged) to format and check files in the staging area before each commit.
```
Comment (giorio94) [inline frontend/README.md:13]:
```suggestion
We use ESLint to enforce linting and code quality. We use Prettier to have a uniform code style. These checks are enforced by a GitHub Action run on PRs. Also locally a pre-commit hook written with [husky](https://typicode.github.io/husky/#/) uses [lint-staged](https://github.com/okonet/lint-staged) to format and check files in the staging area before each commit.
```

File: frontend/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+## Guidelines
+
+We define some coding guidelines to ease the collaboration in our team and to ease the review process.
+
+### Dir structure
+
+The components are in `src/components`. The folder has subdirs, one for each page of the app, plus `misc` for the miscellaneous UI elements (those common between all components) and a `util` dir for custom UI elements used multiple times across the code (custom dialogs, inputs, etc...).
+
+Each component needs to have its own folder with the following structure, e.g. for an `Exmaple` component:
```
Comment (giorio94) [inline frontend/README.md:23]:
```suggestion
Each component needs to have its own folder with the following structure, e.g. for an `Example` component:
```

File: frontend/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+- `Example.tsx` - component definition
+- `Example.stories.ts` - storybook stories to show component usage
+- `Exmaple.css` - css specific code for component
+- `index.ts` - utility file to shorten component import statement in other files
+
+### File structure
+
+Refer to the [Example component] for a demo
```
Comment (giorio94) [inline frontend/README.md:32]: I suppose this should be a link

File: frontend/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+#### Component
+
+Each component file needs to host a single components and needs to have the following structure:
```
Comment (giorio94) [inline frontend/README.md:36]:
```suggestion
Each component file needs to host a single component and needs to have the following structure:
```

File: frontend/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+- import declarations (
+  When you can use non-default imports use them to reduce bundle size)
+- general constant declarations for the components(if needed)
```
Comment (giorio94) [inline frontend/README.md:40]:
```suggestion
- import declarations (when you can use non-default imports use them to reduce bundle size)
- general constant declarations for the components (if needed)
```

File: frontend/README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
+# CrownLabs frontend
+
+This file describes the structure and general coding guidelines of the CrownLabs frontend.
+
+## Structure
+
+Our frontend is a [React](https://it.reactjs.org/) application. We use [antd](https://ant.design/) as the main component library and [Tailwind](https://tailwindcss.com/) utilities to handle specific css scenarios (padding and margin). We also use [Storybook](https://storybook.js.org/) to ease teamwork.
```
Comment (giorio94) [inline frontend/README.md:7]: It may be worth mentioning the choice of typescript instead of javascript

**Review by giorio94 — APPROVED** [review]
/lgtm Thanks @GabriFila

**Comment by qcfe** [comment]
The `package-lock.json` file isn't tracked. It's usually a good thing to include it into the repository in order to ensure dependencies don't break during builds (if I got it right)

**Comment by GabriFila** [comment]
> The `package-lock.json` file isn't tracked. It's usually a good thing to include it into the repository in order to ensure dependencies don't break during builds (if I got it right)

Yes we can add it, I do it now

File: frontend/README.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
+# CrownLabs frontend
+
+This file describes the structure and general coding guidelines of the CrownLabs frontend.
+
+## Structure
+
+Our frontend is a [React](https://it.reactjs.org/) application. We use [antd](https://ant.design/) as the main component library and [Tailwind](https://tailwindcss.com/) utilities to handle specific css scenarios (padding and margin). We also use [Storybook](https://storybook.js.org/) to ease teamwork. We chose to use [Typescript](https://www.typescriptlang.org/) to have a bettere development experience.
```
Comment (giorio94) [inline frontend/README.md:7]:
```suggestion
Our frontend is a [React](https://it.reactjs.org/) application. We use [antd](https://ant.design/) as the main component library and [Tailwind](https://tailwindcss.com/) utilities to handle specific css scenarios (padding and margin). We also use [Storybook](https://storybook.js.org/) to ease teamwork. We chose to use [Typescript](https://www.typescriptlang.org/) to have a better development experience.
```

**Review by qcfe — APPROVED** [review]

**Pull request merged**